scripts/publish.sh

```diff
@@ -21,9 +21,19 @@ git add dist --force
 git commit -m "Added files generated by Travis build"
 
 # Push to releases branch
+EXISTING=`git ls-remote --heads https://github.com/"$TRAVIS_REPO_SLUG".git "$TRAVIS_BRANCH"-dist`
+
 if [ -z "$TRAVIS_TAG" ]
 then
-  git push upstream $TRAVIS_BRANCH:$TRAVIS_BRANCH-dist --force -v
+  if [ -n "$EXISTING" ]
+  then
+    git fetch upstream $TRAVIS_BRANCH-dist:$TRAVIS_BRANCH-dist
+    git checkout $TRAVIS_BRANCH-dist
+    git merge -Xtheirs $TRAVIS_BRANCH --no-edit --ff
+    git push upstream $TRAVIS_BRANCH-dist --force -v
```

Comment:
I believe the reason why the previous commits were lost was because we use a force push. Couldn't we just remove the --force switch instead of merging branches? Technically, there is nothing wrong with merging, but a lot of extra steps?